xskillscore/core/deterministic.py

```diff
@@ -73,7 +79,14 @@ def _determine_input_core_dims(dim, weights):
     return input_core_dims
 
 
-def linslope(a, b, dim=None, weights=None, skipna=False, keep_attrs=False):
+def linslope(
+    a: XArray,
+    b: XArray,
+    dim: Dim = None,
```

Comment:
`dim: Dim | None = None`?

Reply:
In function arguments, you don't have to add `| None` -- it's implied if you make None the default

Reply:
`Dim` contains `None` already